src/tools/perf/perftest.c

```diff
@@ -126,6 +127,10 @@ static int sock_io(int sock, ssize_t (*sock_call)(int, void *, size_t, int),
             ucs_assert(pfd.revents & poll_events);
 
             ret = sock_call(sock, (char*)data + total, size - total, 0);
+            if (!ret && (poll_events == POLLIN)) {
+                ucs_error("%s() remote closed connection", name);
+                exit(EXIT_FAILURE);
```

Comment:
do we need to exit here?

Reply:
now returning failure and moved up exit at sock rte